bindings/pyroot/pythonizations/python/ROOT/pythonization/_roofit/__init__.py

```diff
@@ -30,8 +30,33 @@
     python_classes_dict[python_class.__name__] = python_class
 
 
-def ismagicfunc(name):
-    return name.startswith("__") and name.endswith("__")
+def get_defined_attributes(klass):
+    """
+    Get all class attributes that are defined in a given class or in any of its
+    base classes (except for `object`).
+    """
+
+    blacklist = ["__dict__", "__doc__", "__hash__", "__module__", "__weakref__"]
```

Comment:
Perhaps if you just inspected the methods of the mirror class you wouldn't even need this blacklist?

Reply:
I tried that out, but these few attributes are always in `__dict__`, even if you get it from the class itself. I was trying out some other ways to avoid selecting these attributes, but I didn't find a simple solution that works for both Python 2 and Python 3.

So in the end I thought a short blacklist is more digestible. Or you absolutely want to avoid a hardcoded blacklist?

Reply:
It's not ideal, but if there is no way to tell them apart from your methods, I don't see any other way.